src/pathfinder/opf/opf_ship.cpp

```diff
@@ -195,11 +195,10 @@ Track OPFShipChooseTrack(const Ship *v, TileIndex tile, DiagDirection enterdir,
 	Track track;
 
 	/* Let's find out how far it would be if we would reverse first */
-	Trackdir trackdir = v->GetVehicleTrackdir();
-	TrackBits b = TrackStatusToTrackBits(GetTileTrackStatus(tile2, TRANSPORT_WATER, 0)) & DiagdirReachesTracks(ReverseDiagDir(enterdir)) & TrackdirBitsToTrackBits(TrackdirToTrackdirBits(trackdir));
+	TrackBits b = TrackStatusToTrackBits(GetTileTrackStatus(tile2, TRANSPORT_WATER, 0)) & DiagdirReachesTracks(ReverseDiagDir(enterdir));
 
 	uint distr = UINT_MAX; // distance if we reversed
-	if (b != 0) {
+	if ((b & TrackToTrackBits(TrackdirToTrack(v->GetVehicleTrackdir())) != TRACK_BIT_NONE) {
```

Comment:
The behaviour of this code has changed.
Previously "b" was masked by "trackdir".
Now FindShipTrack gets an less-masked "b".